### PR TITLE
chore(agent-eval): fix `MemorySaver` version compatibility issue

### PR DESCRIPTION
**Pull request by weekenthralling**
After `langgraph-checkpoint>=2.0.5`, `MemorySaver` introduced `ExitStack` to dynamically manage contexts. However, using the with keyword does not work as expected.

In `MemorySaver`, the` __enter__ `method returns an `ExitStack` object, and `ExitStack` does not have a `get_next_version` method.

```python
def __enter__(self) -> "MemorySaver":
        return self.stack.__enter__()
```

This leads to an error when calling `graph.invoke`, error message: "AttributeError: 'ExitStack' object has no attribute 'get_next_version'".

```python
# AsyncPregelLoop
self.checkpointer_get_next_version = checkpointer.get_next_version
```

### CODE REVIEW

File: realtabbench/agent_eval/evaluator.py [hunk 1]
```diff
@@ -66,66 +66,66 @@ async def run_eval(
         if student_context is None:
             student_context = {}
 
-        with MemorySaver() as checkpointer:
-            student = await create_student_graph(
-                datasets=payload.get("datasets"),
-                checkpointer=checkpointer,
-                **student_context,
+        checkpointer = MemorySaver()
```
Comment (edwardzjl) [inline realtabbench/agent_eval/evaluator.py:69]: Why not using the context manager pattern?

Comment (weekenthralling) [reply realtabbench/agent_eval/evaluator.py:69]: In this commit [# 886df0f](https://github.com/langchain-ai/langgraph/commit/886df0fa86841388cbd9bf257733a50555174551), `ExitStack` was introduced. The official explanation is `Context manager for dynamic management of a stack of exit callbacks.`

Comment (edwardzjl) [reply realtabbench/agent_eval/evaluator.py:69]: Explain "doesn't work"

File: pyproject.toml [hunk 2]
```diff
@@ -28,6 +28,8 @@ dependencies = [
   "langchain-core>=0.3.0,<1.0.0",
   "langchain-qdrant>=0.1.4,<1.0.0",
   "langgraph>=0.0.68,<1.0.0",
+  # `MemorySaver` can dynamically manage the `Context Manager` starting from version 2.0.5
+  "langgraph-checkpoint>=2.0.5,<3.0.0",
```
Comment (edwardzjl) [inline pyproject.toml:32]: Are we using this feature in `tablegpt-agent`?

File: realtabbench/agent_eval/evaluator.py (hunk 1 above)
Comment (weekenthralling) [reply realtabbench/agent_eval/evaluator.py:69]: In `MemorySaver`, the` __enter__ `method returns an `ExitStack` object

```python
def __enter__(self) -> "MemorySaver":
        return self.stack.__enter__()
```

But `ExitStack` does not have a `get_next_version` method. This causes the `loop` to attempt calling a method that does not exist when invoking `graph.invoke`.

```python
# AsyncPregelLoop
self.checkpointer_get_next_version = checkpointer.get_next_version
```

**Review by edwardzjl — APPROVED** [review]

**Pull request merged**